test/manual/mongodb_oidc.prose.test.ts

```diff
@@ -12,602 +27,1208 @@ describe('MONGODB-OIDC', function () {
+        it('fails authentication', async function () {
+          try {
+            await collection.findOne();
+            expect.fail('Expected OIDC auth to fail with no user provided');
+          } catch (e) {
+            expect(e).to.be.instanceOf(MongoServerError);
+            expect(e.message).to.include('Authentication failed');
+          }
+        });
+      });
+
+      describe('1.6 Allowed Hosts Blocked', function () {
+        before(function () {
+          cache.clear();
+        });
+
+        // Clear the cache.
+        // Create a client that uses the OIDC url and a request callback, and an
+        // ``ALLOWED_HOSTS`` that is an empty list.
+        // Assert that a ``find`` operation fails with a client-side error.
+        // Close the client.
+        context('when ALLOWED_HOSTS is empty', function () {
           before(function () {
-            client = new MongoClient(
-              'mongodb://test_user1@localhost:27018/?authMechanism=MONGODB-OIDC&directConnection=true&readPreference=secondaryPreferred',
-              {
-                authMechanismProperties: {
-                  REQUEST_TOKEN_CALLBACK: requestCallback
-                }
+            client = new MongoClient('mongodb://localhost/?authMechanism=MONGODB-OIDC', {
+              authMechanismProperties: {
+                ALLOWED_HOSTS: [],
+                REQUEST_TOKEN_CALLBACK: createRequestCallback('test_user1', 600)
               }
-            );
+            });
             collection = client.db('test').collection('test');
           });
 
-          after(async () => {
-            client?.close();
-          });
-
-          // - Perform a ``find`` operation.
-          // - Clear the cache.
-          it('successfully authenticates', async function () {
-            const doc = await collection.findOne();
-            expect(doc).to.equal(null);
+          it('fails validation', async function () {
+            try {
+              await collection.findOne();
+            } catch (error) {
+              expect(error).to.be.instanceOf(MongoInvalidArgumentError);
+              expect(error.message).to.include(
+                'is not valid for OIDC authentication with ALLOWED_HOSTS'
+              );
+            }
           });
         });
 
-        context('when using aws', function () {
-          const testTokenFile = process.env.AWS_WEB_IDENTITY_TOKEN_FILE;
-          let client;
-          let collection;
+        // Create a client that uses the url ``mongodb://localhost/?authMechanism=MONGODB-OIDC&ignored=example.com`` a request callback, and an
+        // ``ALLOWED_HOSTS`` that contains ["example.com"].
+        // Assert that a ``find`` operation fails with a client-side error.
+        // Close the client.
+        context('when ALLOWED_HOSTS does not match', function () {
+          beforeEach(function () {
+            this.currentTest.skipReason = 'Will fail URI parsing as ignored is not a valid option';
```

Comment:
I don't follow the skipReason here, is this something we will be fixing or innate to our driver? like how we throw instead of warn on uri options?

Reply:
We throw for invalid options in our driver - so this prose test is skipped and will never be unskipped. I can remove the test entirely but wanted it to be shown that the test itself was not missed.

Reply:
Yea that makes sense to me, ty for clarifying